config.go

```diff
@@ -77,9 +78,22 @@ func upgradeConfig(cfg *Config, absFilePath string) error {
 
 // WriteConfig saves a devbox config file.
 func WriteConfig(path string, cfg *Config) error {
+	err := cfg.validate()
+	if err != nil {
+		return err
+	}
 	return cuecfg.WriteFile(path, cfg)
 }
 
+func (c *Config) validate() error {
+	for k := range c.Shell.Scripts {
+		if k == "" {
+			return errors.New("cannot have script with empty name in devbox.json")
+		}
+	}
+	return nil
```

Comment:
oh boy, I ended up adding another validate function today, so will need to merge it. It should be fairly straightforward.

@loreto had requested me to see if we can use the `cue` validation instead, which I am yet to do.

Reply:
Yeah, I actually attempted to use `cue` for validation, but I couldn't find what I needed quickly enough so I gave up and added the validation function. I can merge it with yours

Reply:
done